orchestra/bots/staffbot.py

```diff
@@ -63,19 +70,45 @@ def _send_task_to_workers(self, task, required_role):
                 self._send_task_to_worker(
                     worker, task, required_role)
 
-    def _send_task_to_worker(self, worker, task):
+    def _send_task_to_worker(self, worker, task, request_cause):
         """
         Send the task to the worker for them to accept or reject.
         """
-        staffing_request = StaffingRequest.objects.create(
-            worker=worker,
-            task=task,
-            request_cause=StaffingRequest.RequestCause.AUTOSTAFF.value,
-            communication_method=(
-                StaffingRequest.CommunicationMethod.EMAIL.value))
+        communication_type = (
+            CommunicationPreference.CommunicationType.NEW_TASK_AVAILABLE.value)
+
+        communication_preference = CommunicationPreference.objects.get(
+            communication_type=communication_type,
+            worker=worker)
+
+        if communication_preference.can_email():
```

Comment:
it might be cleaner to just pass the preference in and perform the check inside the send function since we'll probably want to reuse the logic